sockeye/arguments.py

```diff
@@ -820,10 +820,26 @@ def add_training_args(params):
                               help='How to normalize the loss. By default loss is normalized by the number '
                                    'of valid (non-PAD) tokens (%s).' % C.LOSS_NORM_VALID)
 
+    train_params.add_argument('--lenratio-link',
+                              type=str,
+                              default='none',
+                              choices=['none', 'normal', 'poisson'],
+                              help='Train with prediction of the ratio of reference length to input length. Default %(default)s.'
+                              'using the specified links function. Default %(default)s.')
+    train_params.add_argument('--lenratio-layers',
+                              type=int,
+                              default=1,
+                              choices=[1, 2],
+                              help='Number of layers for predicting the length ratio. Default %(default)s.')
```

Comment:
It becomes complicated to handle them, because we need to save and to pass all the internal layers' weights to initialize the length ratio prediction block during inference. Maybe we could add more later, after everything else is settled.

Reply:
Why do we need the internal layers' weights in inference anyway? The reason we have the output layer weights optionally pulled out at inference time is the vocabulary selection feature, where you need to cut the computron graph of the decoder short to do a custom softmax over a reduced set of symbols.
However, your case is independent from that. You can just the symbolic definition of the length prediction subnetwork to the encoder module in the inference model and get the result.

Reply:
True, implemented it all in symbols; actually shortened the implementation a bit. And can now support arbitrary number of FF layers.